main.py

```diff
@@ -648,23 +648,105 @@ def on_link_down(self, event):
 
     def handle_link_down(self, event):
         """Change circuit when link is down or under_mantenance."""
-        log.debug("Event handle_link_down %s", event)
+        link = event.content["link"]
+        log.info("Event handle_link_down %s", link)
+        switch_flows = {}
+        evcs_with_failover = []
+        evcs_normal = []
+        check_failover = []
         for evc in self.circuits.values():
+            if evc.is_affected_by_link(link):
+                # if there is no failover path, handles link down the
+                # tradditional way
+                if (
+                    not getattr(evc, 'failover_path', None) or
+                    evc.is_failover_path_affected_by_link(link)
+                ):
+                    evcs_normal.append(evc)
+                    continue
+                for dpid, flows in evc.get_failover_flows().items():
+                    switch_flows.setdefault(dpid, [])
+                    switch_flows[dpid].extend(flows)
+                evcs_with_failover.append(evc)
+            else:
+                check_failover.append(evc)
+
+        offset = 0
+        while switch_flows:
```

Comment:
In the future, it might worth extracting this batched/waited event emission as a function to be utilized, but only when a next NApp needs this. Very neat solution.